### ISSUE TITLE
Post listing grid view boxes too big

### ISSUE THREAD

**Issue by marvinkome** [decaporg/decap-cms#1341]
**- Do you want to request a *feature* or report a *bug*?**
Feature

**- What is the current behaviour?**
Grid boxes for post listing are too big if they only display title

**- If the current behaviour is a bug, please provide the steps to reproduce.**

**- What is the expected behaviour?**
Boxes should add more information about the post or become smaller

**Comment by erquhart** [comment]
Yeah, they were never supposed to stay that way, more info should definitely be shown in the grid boxes - same goes for list view, too.

**Comment by marvinkome** [comment]
What info should be there. I can pick it up (I mean make a PR)

**Comment by erquhart** [comment]
Awesome! We still need design, so I'm not certain this is ready to roll, but feel free to take a swing. We'll want to use inferred fields, which try to smartly determine things like author, date, etc, for each entry.

There's a [set of constants](https://github.com/netlify/netlify-cms/blob/c31aaae0d4dc02daa60979b134e2aa8b214aece0/src/constants/fieldInference.js) that show what fields are being inferred, and you can see how inferred fields are already used to get the entry title here:

https://github.com/netlify/netlify-cms/blob/1d4162050956a25437882f2423f5b74a68c1c926/src/components/Collection/Entries/EntryCard.js#L22

**Comment by marvinkome** [comment]
Is there a structure of how it should be displayed and what content should be inside? Also, the inferred constants don't have a date field, that would be useful in the boxes.

**Comment by erquhart** [comment]
> Is there a structure of how it should be displayed and what content should be inside?

That's why I mentioned the need for design here. As far as what to display, I'd just go with the available inferred fields, namely `author`, `description`, and `image`.

> Also, the inferred constants don't have a date field, that would be useful in the boxes.

Agreed, but I'd recommend opening a separate issue and handling that in a different PR to keep this moving. Determining what fields to infer the date from and how best to display it would slow this PR down unnecessarily.

**Comment by mittalyashu** [comment]
> As far as what to display, I'd just go with the available inferred fields, namely author, description, and image.

Showing image in that small box would be hard, according to design.

![image](https://user-images.githubusercontent.com/29014463/44960182-99403d00-af18-11e8-95f8-3e6427774391.png)

We can show content like `text` or `avatar image`.

**Comment by erquhart** [comment]
See old grid card design with images here: https://0-7-6--cms-demo.netlify.com/#/collections/posts

**Comment by mittalyashu** [comment]
According to this design

![image](https://user-images.githubusercontent.com/29014463/45049863-85721380-b09d-11e8-9123-938b3fc470ae.png)

We have:

1. Image _(media uploader widget)_
2. Title _(string widget)_
3. Date _(date widget)_
4. Body _(markdown widget)_

and sometimes a user might add more widgets to the collection and showing all the content on the gird can get messy.

Is there some rules or prioritization for NetlifyCMS that only selected content will be shown?

**Comment by erquhart** [comment]
No, back then we just showed everything but cut off the overflow.

**Comment by mittalyashu** [comment]
> we just showed everything but cut off the overflow.

Hmm... 🤔

What if we design every element which will be shown on the grid or list view, but we will add the option where users can select which content they want to see, the same concept can be used for list and grid. view.

**Comment by erquhart** [comment]
That's always been the plan, just hasn't been a priority.

**Comment by charlie-s** [comment]
Anyone working on this? Would really like to see image preview in my collections.

**Comment by tomrutgers** [comment]
It's currently not claimed by anyone

**Comment by erquhart** [comment]
@charlie-s we'd love the contribution - comment here or in [Gitter](https://gitter.im/netlify/netlifycms) if you have any questions while checking this out.

**Comment by coldice** [comment]
Following up on @mittalyashu comment, how about just a list of field names to be shown? Or has this box more config options already?

**Comment by martinjagodic** [comment]
This was addressed a while ago — the grid (and list) entry cards now render the inferred image, the entry title, and a workflow status badge instead of just an oversized empty-looking box.
